### PR TITLE
Add script for migrating old subject colors to new subject colors

### PR DESCRIPTION
**Pull request by bungaepyo**
### Summary 

This pull request adds a script to migrate all old subject colors to new subject colors (Gray, Red, Orange, Green, Blue, Purple). Since we got rid of **Gray**, I migrated it to **Ithaca Winter**. This migration will make changes to color values in the user-subject-colors collection, but won't affect the colors of courses already added to the semesters. For now, I have only run single user dry-runs on dev.

- [x] added color migration script

Extends #573 
Reference from #593 

### Test Plan 
Screen Shot 2021-12-04 at 5 41 37 PM

`npm run ts-node -- src/requirements/admin/color-migration <user email>` to dry-run on a single user.
`npm run ts-node -- src/requirements/admin/color-migration --run-on-db` to run on the entire database.

### CODE REVIEW

**Comment by hahnbeelee** [comment]
Can you push the console logs that you have in that screenshot? It's hard to tell what's happening without them

**Comment by hahnbeelee** [comment]
LGTM - feel free to run on the dev db instance and check if everything looks good after it!

File: src/requirements/admin/color-migration.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+async function main() {
+  let userEmail = process.argv[2];
+  const runOnDB = process.argv.includes('--run-on-db');
+  if (userEmail != null && userEmail !== '--run-on-db') {
+    await runOnUser(userEmail, runOnDB);
+    return;
+  }
+  const collection = await usernameCollection.get();
```
Comment (benjamin-shen) [inline src/requirements/admin/color-migration.ts:64]: Can you run this on every doc in the `subjectColorsCollection`? 
```suggestion
  const collection = await subjectColorsCollection.get();
```

**Review by benjamin-shen — COMMENTED** [review]
Is it possible that non-onboarded users have old data in the colors collection?

**Comment by hahnbeelee** [comment]
Ran on dev db instance successfully.

**Review by hahnbeelee — APPROVED** [review]
Ran on dev db instance successfully.

**Review by benjamin-shen — APPROVED** [review]

**Pull request merged**